guardian/typerighter
# Use CDK, not cloudformation, for all apps

## What does this change?

Moves all our apps to the latest version of CDK. This involves a few high-level changes:

- Migrating the checker app to CDK
- Using the latest Patterns where they're available
- Using CDK for our DNS
- Making the appropriate changes to our deployment

We think it'll be easier to spin up a new stack in parallel to the old, and then point consumer apps to the new subdomains. We've taken this opportunity to rationalise the subdomains:

`(api. -> checker.)typerighter`
`({no subdomain} -> manager.)typerighter`

## How to test

This PR is ... larger than I'd like 😅 There wasn't an easy way to make this change incrementally, so I've gone for a single PR. I'd be super happy to pair on a review.

Deploy the app to CODE. You should see services on the following subdomains behaving as expected:

`checker.typerighter.code.dev-gutools.co.uk`
`manager.typerighter.code.dev-gutools.co.uk`

To exercise the service, change the configurating in a consuming service, e.g. Composer CODE, to point to the new URL. You should see it working correctly.

Tested in Composer CODE. This is currently deployed, so you should be able to test as-is.

## How can we measure success?

- The app continues to work as expected
- CDK! Fewer LOC, clearer configuration.

## Have we considered potential risks?

There shouldn't be any denial of service, as we're pointing consuming services to a new URL.
